x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/risk_score_data_client.test.ts

```diff
@@ -452,4 +454,37 @@ describe('RiskScoreDataClient', () => {
       }
     });
   });
+  describe('upgrade process', () => {
+    beforeEach(() => {
+      jest.spyOn(riskScoreDataClient, 'upgrade');
+      spyOnPrivateMethod(riskScoreDataClient, 'upsertRiskScoreLatestIndex');
+    });
+    it('upserts the configuration for the latest risk score index when upgrading', async () => {
+      await riskScoreDataClient.upgrade();
+
+      expect(
+        getPrivateMethod(riskScoreDataClient, 'upsertRiskScoreLatestIndex')
+      ).toHaveBeenCalled();
+    });
+    it('upserts the configuration for the latest risk score index only a single time, no matter how many times upgrade is called', async () => {
+      await Promise.all(
+        times(3, () => {
+          return riskScoreDataClient.upgrade();
+        })
+      );
+      expect(
+        getPrivateMethod(riskScoreDataClient, 'upsertRiskScoreLatestIndex')
+      ).toHaveBeenCalledTimes(1);
+    });
+  });
 });
+
+const getPrivateMethod = (target: unknown, method: string) => {
```

Comment:
I _really_ liked this approach, so that I don't have to make a method public just to spy on it. However, I am willing to change this if anyone has found a better approach!

Reply:
Done: https://github.com/elastic/kibana/pull/174474/commits/f60cb153b190d3e4abab4f2412c44d53cb65a198